### PR TITLE
JAPI-573 Add in default support for set of

### PR DESCRIPTION
**Pull request by jchambers-ln**
https://track.hpccsystems.com/browse/JAPI-573

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [x] This change does not cause any existing JUnits to fail.
  - [x] I have include JUnit coverage to test this change
  - [x] I have performed system test and covered possible regressions and side effects.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] This change fixes the problem, not just the symptom

## Testing:
Basic unit test added to hpcc4j, additional testing added to DSP to validate it works with the requirements leading to it being added.

### CODE REVIEW

**Comment by jchambers-ln** [comment]
I was also able to rework the antlr to support all of the following

RECORD
STRING1 test;
SET OF UTF8 fieldAllset {DEFAULT (ALL)};
SET OF UTF8 field2set {DEFAULT (['test1','test2'])};
SET OF UTF8 field3set {DEFAULT ([''])};
SET OF UTF8 field4set {DEFAULT (['test'])};
STRING2 test;
END;

**Comment by rpastrana** [comment]
Test failures seem unrelated to proposed changes.

**Review by rpastrana — APPROVED** [review]
@jchambers-ln it looks fine, but would like @drealeed and @jpmcmu to chime in as well

**Review by drealeed — APPROVED** [review]
Looks good to me

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecordReader.java [hunk 1]
```diff
@@ -396,6 +396,68 @@ else if (currentrec != null)
...
+            if(val.substring(0, 2).equals("['")) {
+                val = val.substring(2, val.length());
+            }
+        }
+        if(val.length() >= 2){
+            if(val.substring(val.length()-2, val.length()).equals("']")) {
+                val = val.substring(0, val.length() - 2);
+            }
+        }
+        //val = val.replaceFirst("(\[')", "");
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecordReader.java:437]: Minor: formatting differs here, and might want to remove these commented out lines.

Comment (jchambers-ln) [reply wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecordReader.java]: removed commented code

**Pull request merged**